### PR TITLE
Move Unreal Implementation to OpenPype

### PR DESCRIPTION
**Pull request by antirotor**
## Changes

This is moving host specific logic from Avalon to OpenPype. It is removing integration plugin as submodule and adding it directly to Unreal host implementation. That plugin was "rebranded" to OpenPype.

To test successful move, create new Unreal project and go over basic functions to ensure it is working.

### CODE REVIEW

**Comment by mkolar** [comment]
Task linked: [OP-2246 Consume `avalon-unreal-integration` plugin and rebrand](https://app.clickup.com/t/6658547/OP-2246)

File: openpype/hosts/unreal/api/__init__.py [hunk 1]
```diff
@@ -15,8 +35,28 @@
...
+    "show_manager",
+    "show_experimental_tools",
+    "show_tools_dialog",
+    "show_tools_popup",
+    "instantiate"
+]
+
+
+
 def install():
```
Comment (iLLiCiTiT) [inline openpype/hosts/unreal/api/__init__.py:58]: Content of this `install` (and `uninstall`) should be in pipeline.py.

File: openpype/hosts/unreal/__init__.py [hunk 2]
```diff
@@ -3,11 +3,12 @@
 
 def add_implementation_envs(env, _app):
     """Modify environments to contain all required for implementation."""
-    # Set AVALON_UNREAL_PLUGIN required for Unreal implementation
+    # Set OPENPYPE_UNREAL_PLUGIN required for Unreal implementation
     unreal_plugin_path = os.path.join(
-        os.environ["OPENPYPE_REPOS_ROOT"], "repos", "avalon-unreal-integration"
+        os.environ["OPENPYPE_ROOT"], "openpype", "hosts",
```
Comment (iLLiCiTiT) [inline openpype/hosts/unreal/__init__.py:8]: This would be better if it is relative to current file.

**Comment by simonebarbieri** [comment]
I'm getting this error when trying to launch Unreal from a new task (no existing project for the task):

```
WARNING:openpype.lib.python_module_tools:Failed to load path: "C:\Dev\pype\simonebarbieri-pype\openpype\hosts\unreal\hooks\pre_workfile_preparation.py"
Traceback (most recent call last):
  File "C:\Dev\pype\simonebarbieri-pype\openpype\lib\python_module_tools.py", line 91, in modules_from_path
    module = import_filepath(full_path, mod_name)
  File "C:\Dev\pype\simonebarbieri-pype\openpype\lib\python_module_tools.py", line 36, in import_filepath
    module_loader.exec_module(module)
  File "<frozen importlib._bootstrap_external>", line 728, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "C:\Dev\pype\simonebarbieri-pype\openpype\hosts\unreal\hooks\pre_workfile_preparation.py", line 13, in <module>
    from openpype.hosts.unreal.api import lib as unreal_lib
  File "C:\Dev\pype\simonebarbieri-pype\vendor\python\shiboken2\files.dir\shibokensupport\__feature__.py", line 142, in _import
    return original_import(name, *args, **kwargs)
  File "C:\Dev\pype\simonebarbieri-pype\openpype\hosts\unreal\api\__init__.py", line 11, in <module>
    from .pipeline import (
  File "C:\Dev\pype\simonebarbieri-pype\vendor\python\shiboken2\files.dir\shibokensupport\__feature__.py", line 142, in _import
    return original_import(name, *args, **kwargs)
  File "C:\Dev\pype\simonebarbieri-pype\openpype\hosts\unreal\api\pipeline.py", line 5, in <module>
    import unreal  # noqa
  File "C:\Dev\pype\simonebarbieri-pype\vendor\python\shiboken2\files.dir\shibokensupport\__feature__.py", line 142, in _import
    return original_import(name, *args, **kwargs)
ModuleNotFoundError: No module named 'unreal'
```

File: openpype/hosts/unreal/integration/.gitignore [hunk 3]
```diff
@@ -0,0 +1,35 @@
+# Prerequisites
```
[the remark this answers is not in the document]
Comment (antirotor) [reply openpype/hosts/unreal/integration/.gitignore:1]: This is just adding c++ and unreal specific stuff locally, it doesn't override global `.gitignore`

[the remark this answers is not in the document]
Comment (iLLiCiTiT) [reply openpype/hosts/unreal/integration/.gitignore:1]: So it is overriding it :) (for this folder)

**Review by simonebarbieri — APPROVED** [review]
I've tried creating a new project, loading and updating some assets, and it seems to be working fine!

**Pull request merged**